harness/tests/storage/util.py

```diff
@@ -108,6 +108,10 @@ def run_storage_lifecycle_test(
         with pytest.raises(errors.CheckpointNotFound):
             with manager.restore_path(storage_id) as path:
                 pass
+
+        # Ensure second delete does not puke.
+        manager.delete(storage_id)
```

Comment:
probably one of these additions to the lifecycle test is sufficient, the others are just noise.